packages/react-native-client/android/src/main/java/org/membraneframework/reactnative/VideoRendererView.kt

```diff
@@ -16,19 +26,39 @@ class VideoRendererView(
   RNFishjamClient.OnTrackUpdateListener {
   var activeVideoTrack: VideoTrack? = null
   var trackId: String? = null
+  private var mirrorVideo: Boolean = false
+  private val coroutineScope = CoroutineScope(Dispatchers.Main + Job())
 
   private val videoView =
     RNFishjamClient.fishjamClient.createVideoViewRenderer().also {
       addView(it)
       RNFishjamClient.onTracksUpdateListeners.add(this)
     }
 
+  private var fadeAnimation: ValueAnimator =
+    ValueAnimator.ofArgb(Color.TRANSPARENT, Color.BLACK).apply {
+      duration = 100
+      interpolator = LinearInterpolator()
+      addUpdateListener {
+        val colorValue = it.animatedValue as Int
+        foreground = ColorDrawable(colorValue)
+      }
+    }
+
   private fun setupTrack(videoTrack: VideoTrack) {
     if (activeVideoTrack == videoTrack) return
 
     activeVideoTrack?.removeRenderer(videoView)
-    videoTrack.addRenderer(videoView)
     activeVideoTrack = videoTrack
+
+    if (videoTrack is LocalVideoTrack && videoTrack.capturer is CameraCapturer) {
+      (videoTrack.capturer as CameraCapturer).setMirrorVideo = { isFrontCamera ->
+        setMirrorVideo(isFrontCamera)
+      }
+      setMirrorVideo(getEnumerator(context).isFrontFacing((videoTrack.capturer as CameraCapturer).cameraName))
+    }
+
+    videoTrack.addRenderer(videoView)
```

Comment:
this is the same code as in Preview, can we reuse it somehow? or add it to the VideoViewRenderer from android client?

Reply:
`VideoTextureViewRenderer` does not have capturer as parameter. So it would not be possible to determine camera properties.

Reply:
but you have video track
what if video track had a method `setMirrorVideo`? and the video track would interact with its private capturer?

Reply:
VideoTextureViewRendered does not have videotrack, correct me if i am mistaken